Add bucket-based recipe factory w/ example

CSX8600 opened the pull request.

CSX8600 committed:
Add bucket-based recipe factory w/ example

RavenholmZombie commented on src/main/resources/assets/milky/recipes/bucket_chocolate_milk.json:
```diff
@@ -0,0 +1,16 @@
+    "type": "milky:milk_based",
+    "ingredients": [
+        {
+			"item": "minecraft:milk_bucket"
```
This should be the Forge Universal Bucket containing the mod's Milk fluid, not the vanilla Milk Bucket item.

CSX8600 replied:
done

CSX8600 committed:
Added bucket-based ingredient factory

Also made clear that the milk-based factory is actually a Shapeless Bucket Result factory
